h2o-py/h2o/automl/autoh2o.py

```diff
@@ -144,99 +144,89 @@ def __init__(self,
                   "*******************************************************************\n" \
                   "\nVerbose Error Message:")
 
-        
-        # Make bare minimum build_control (if max_runtimes_secs is an invalid value, it will catch below)
-        self.build_control = {
-            'stopping_criteria': {
-                'max_runtime_secs': max_runtime_secs,
-            }
-        }
-
-        # Make bare minimum build_models
-        self.build_models = {
-            'exclude_algos': None
-        }
-
-        # nfolds must be an non-negative integer and not equal to 1:
-        if nfolds is not 5:
-            assert_is_type(nfolds,int)
+        self._job = None
+        self._leader_id = None
+        self._leaderboard = None
+        self._verbosity = verbosity
+        self._event_log = None
+        self._training_info = None
+        self._state_json = None
+
+        # Make bare minimum params containers
+        self.build_control = dict(
+            stopping_criteria=dict()
+        )
+        self.build_models = dict()
+        self.input_spec = dict()
+
+        # build_control params #
```

Comment:
standardized and simplified parameters validation in this file:
- removed special handling for enums (stopping metric and sort_metric).
- removed special handling of default values.
- all params follow the same logic
 1. assert type
 2. additional validation only if necessary
 3. pass value to backend
backend handles all conversions, case sensitivity...whatever.